### PR TITLE
Fix support for relative path descriptions to root files in readRNOGDataMattak

### PR DESCRIPTION
**Pull request by sjoerd-bouma**
The mattak reader currently only supports reading directories. Support for reading in `.root` files in readRNOGDataMattak is implemented by symlinking to a temporary directory. However, it seems that passing a relative path to `os.symlink` doesn't behave as expected. This is resolved here by explicitly calling `os.path.abspath(dir_file)`.

Also makes some minor improvements to the documentation and makes some of the module functions protected, to reduce the clutter in the online documentation.

### CODE REVIEW

File: NuRadioReco/modules/io/RNO_G/readRNOGDataMattak.py [hunk 1]
```diff
@@ -774,7 +809,7 @@ def get_event_by_index(self, event_index):
         Returns
         -------
 
-        evt: NuRadioReco.framework.event
+        evt: `NuRadioReco.framework.event.Event`
```
Comment (fschlueter) [inline NuRadioReco/modules/io/RNO_G/readRNOGDataMattak.py:812]: out of curiosity, why?

**Review by fschlueter — APPROVED** [review]
Thanks for doing this @sjoerd-bouma. Should we make a minor release with this fix/feature.

File: NuRadioReco/modules/io/RNO_G/readRNOGDataMattak.py (hunk 1 above)
Comment (sjoerd-bouma) [reply NuRadioReco/modules/io/RNO_G/readRNOGDataMattak.py:812]: The former renders as text, the latter (online) as a clickable reference to the class.

**Comment by fschlueter** [comment]
another thing. the file has a lot of whitespaces (from me). Should we delete them?

**Comment by sjoerd-bouma** [comment]
> Thanks for doing this @sjoerd-bouma. Should we make a minor release with this fix/feature.

Yes, I should update the version number and changelog - I guess we can release this as 2.2.1, unless there are other hotfixes we want to include. Simultaneous releases to `master` and `develop` are still a bit of a hassle, maybe at some point we should think about how to simplify that.

> another thing. the file has a lot of whitespaces (from me). Should we delete them?

No strong feelings there from me, my IDE deletes them only from lines I edit. Up to you.

**Comment by fschlueter** [comment]
> > Thanks for doing this @sjoerd-bouma. Should we make a minor release with this fix/feature.
> 
> Yes, I should update the version number and changelog - I guess we can release this as 2.2.1, unless there are other hotfixes we want to include. Simultaneous releases to `master` and `develop` are still a bit of a hassle, maybe at some point we should think about how to simplify that.
> 
> > another thing. the file has a lot of whitespaces (from me). Should we delete them?
> 
> No strong feelings there from me, my IDE deletes them only from lines I edit. Up to you.

What IDE do you use? I tired to configure my VS code to do that (with an extra plugin) but it does not work for me.

**Comment by sjoerd-bouma** [comment]
> > > Thanks for doing this @sjoerd-bouma. Should we make a minor release with this fix/feature.
> > 
> > 
> > Yes, I should update the version number and changelog - I guess we can release this as 2.2.1, unless there are other hotfixes we want to include. Simultaneous releases to `master` and `develop` are still a bit of a hassle, maybe at some point we should think about how to simplify that.
> > > another thing. the file has a lot of whitespaces (from me). Should we delete them?
> > 
> > 
> > No strong feelings there from me, my IDE deletes them only from lines I edit. Up to you.
> 
> What IDE do you use? I tired to configure my VS code to do that (with an extra plugin) but it does not work for me.

VS Code, with the [Trailing Spaces](https://marketplace.visualstudio.com/items?itemName=shardulm94.trailing-spaces) extension. To delete trailing spaces from modified lines only, add
```
"trailing-spaces.deleteModifiedLinesOnly": true,
"trailing-spaces.trimOnSave": true
```
to your VSCode settings.json.

**Comment by fschlueter** [comment]
> > > > Thanks for doing this @sjoerd-bouma. Should we make a minor release with this fix/feature.
> > > 
> > > 
> > > Yes, I should update the version number and changelog - I guess we can release this as 2.2.1, unless there are other hotfixes we want to include. Simultaneous releases to `master` and `develop` are still a bit of a hassle, maybe at some point we should think about how to simplify that.
> > > > another thing. the file has a lot of whitespaces (from me). Should we delete them?
> > > 
> > > 
> > > No strong feelings there from me, my IDE deletes them only from lines I edit. Up to you.
> > 
> > 
> > What IDE do you use? I tired to configure my VS code to do that (with an extra plugin) but it does not work for me.
> 
> VS Code, with the [Trailing Spaces](https://marketplace.visualstudio.com/items?itemName=shardulm94.trailing-spaces) extension. To delete trailing spaces from modified lines only, add
> 
> ```
> "trailing-spaces.deleteModifiedLinesOnly": true,
> "trailing-spaces.trimOnSave": true
> ```
> 
> to your VSCode settings.json.

I have exactly that too ...

**Review by fschlueter — APPROVED** [review]

**Comment by fschlueter** [comment]
@sjoerd-bouma, any reason to not merge this?

**Comment by sjoerd-bouma** [comment]
> @sjoerd-bouma, any reason to not merge this?

I thought of one! Should be good to merge now though.

**Review by fschlueter — APPROVED** [review]

**Comment by fschlueter** [comment]
ping

**Pull request merged**